demo/api/src/importer/entities/base-import-target.entity.ts

```diff
@@ -0,0 +1,43 @@
+import { BaseEntity, Constructor, Entity, PrimaryKey, Property } from "@mikro-orm/core";
+import { Field, ID, ObjectType } from "@nestjs/graphql";
+import { v4 } from "uuid";
+
+export interface BaseImportInterface {
+    id: string;
+    createdAt?: Date;
+    updatedAt?: Date;
+    deletedAt?: Date;
+}
+
+export type ImporterEntityClass = Constructor;
+
+@ObjectType({ isAbstract: true, description: "Base import entity" })
+@Entity({ abstract: true })
+export abstract class BaseImportTargetEntity<Entity extends object, Primary extends keyof Entity>
```

Comment:
This doesn't work correctly with the API Generator (see the failing workflow) since the generator requires an `id` field. Wouldn't it be better to remove the base entity altogether and implement the interface in the product entity?